test/CometWrapper.t.sol

```diff
@@ -860,6 +869,14 @@ contract CometWrapperTest is BaseTest, CometMath {
         assertEq(cometWrapper.allowance(alice, bob), 100e6);
         vm.stopPrank();
     }
+
+    function calculateAssetsToWithdrawFromShares(uint256 shares) internal returns (uint256) {
+        uint256 currentPrincipal = cometWrapper.totalSupply();
```

Comment:
The names that this repo uses are strange. We have both `assets` and `shares`, and then `totalSupply` and `totalAssets`.

I'm surprised that `currentPrincipal` is `totalSupply`; I would assume that `currentPrincipal` would represent an internal balance that you would need to multiply by something in order to get the `totalSupply` (an external value).

what do `totalSupply` and `totalAssets` represent?

Reply:
Synced offline. `assets/shares` are ERC-4626 terms.